### ISSUE TITLE
Error notfications in archive page and redirecting

### ISSUE THREAD

**Issue by fesarlis** [woothemes/woocommerce#51399]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

I have noticed two strange behavioural issues in WC. Tried with several themes, with no plugins/functions, they still are reproduced.

### Expected behavior

The above issues have an obvious expected behaviour. 

### Actual behavior

 when an error occurs in archive page (eg add a second item in cart when only 1 is in stock), WC automatically redirects to single product page in question and displays the error notification there. The relevant filter (woocommerce_cart_redirect_after_error) which disables redirecting does not work when AJAX add to cart option in WC is enabled, it causes hanging. Also, after redirecting, if we hit 'Back' on browser, you will see the 'Add to cart' button of the product in question disabled and stuck in spinner state.

### Steps to reproduce

.

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.3.1
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.1
Log Directory Writable: ✔
WP Version: 6.6.2
WP Multisite: –
WP Memory Limit: 1 GB
WP Debug Mode: –
WP Cron: ✔
Language: el
External object cache: –

### Server Environment ###

Server Info: 
PHP Version: 7.4.27
PHP Post Max Size: 512 MB
PHP Time Limit: 600
PHP Max Input Vars: 10000
cURL Version: 7.61.1
OpenSSL/1.1.1k

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.6.19-MariaDB-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

ame_ac_changeset: 6
attachment: 973
blocks: 4
custom_css: 1
customize_changeset: 354
filter-field: 3
filter-seo-rule: 3
filter-set: 1
itsec-dash-card: 6
itsec-dashboard: 1
nav_menu_item: 74
oembed_cache: 4
page: 8
post: 1
product: 869
product_variation: 21
revision: 245
snitch: 434
tm_global_cp: 2
view: 7
view-template: 5
wp_global_styles: 1
wp_navigation: 1
wp-types-group: 3
wp-types-term-group: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (44) ###

Query Monitor: by John Blackbourn – 3.16.4
Admin Columns Pro: by AdminColumns.com – 6.4.11
Admin Menu Editor Pro: by Janis Elsts – 2.26
AME Branding Add-on: by Janis Elsts – 1.3.7
The SEO Framework: by The SEO Framework Team – 5.0.6
Breadcrumb NavXT: by John Havlik – 7.3.1
Custom Post Type Cleanup: by keesiemeijer – 1.3.2
Database Cleaner (Pro): by Jordy Meow – 1.1.3
Disable Author Archives: by freemp – 1.3.5
Disable Comments: by WPDeveloper – 2.4.6
Disable Bloat for WordPress & WooCommerce PRO: by Disable Bloat – 3.4.4
Login/Signup Popup: by XootiX – 2.7.6
EWWW Image Optimizer: by Exactly WWW – 7.9.0
Payment Gateway Plugin for PayPal WooCommerce ( Free ): by WebToffee – 1.8.6
Filter Everything PRO: by Andrii Stepasiuk – 1.8.6
Flexible Shipping PRO: by Octolize – 2.18.4
Flexible Shipping: by Octolize – 4.25.8
Font Awesome: by Font Awesome – 4.5.0
Force Regenerate Thumbnails: by Exactly WWW – 2.1.4
Free Shipping Label: by Devnet – 3.1.1
HTML Editor Syntax Highlighter: by James Bradford – 2.4.4
Solid Security Pro: by SolidWP – 8.5.0
Kadence WooCommerce Email Designer: by Kadence WP – 1.5.13
Loco Translate: by Tim Whitlock – 2.6.11
Log cleaner for Solid Security: by Rocket Apps – 1.4.1
Quform: by ThemeCatcher – 2.20.0
Smart Attachment Page Remove: by Peter Raschendorfer – 4.0.5
SMTP Mailer: by naa986 – 1.1.15
Snitch: by pluginkollektiv – 1.2.0
Toolset Types: by OnTheGoSystems – 3.5.2
UberMenu 3 - The Ultimate WordPress Mega Menu: by Chris Mavricos
SevenSpark – 3.8.5

User Role Editor Pro: by Vladimir Garagulia – 4.64.2
Widget Options: by Widget Options Team – 4.0.6.1
WooCommerce Added To Cart Popup: by WeCreativez – 1.4.1
Checkout Field Editor for WooCommerce (Pro): by ThemeHigh – 3.6.3
Extra Product Sorting Options for WooCommerce: by SkyVerge – 2.10.0
Extra Product Options & Add-Ons for WooCommerce: by ThemeComplete – 6.4.6
Toolset WooCommerce Blocks: by OnTheGoSystems – 3.1.6
WooCommerce: by Automattic – 9.3.1
WP All Import Pro: by Soflyy – 4.8.0
WP Rocket: by WP Media – 3.16.4
AME Toolbar Editor: by Janis Elsts – 1.5.1
Toolset Views: by OnTheGoSystems – 3.6.16
WP All Import - WooCommerce Import Add-On Pro: by Soflyy – 3.3.1

### Inactive Plugins (3) ###

Custom Product Tabs for WooCommerce: by Code Parrots – 1.8.5
Custom Product Tabs WP All Import Add-on: by YIKES
Inc. – 2.0.5

Disable payment method / COD fees for Order for WooCommerce: by PI Websolution – 1.1.7.61

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
db.php: Query Monitor Database Class (Drop-in)

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### Logging ###

Enabled: –
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 1 KB

### WC Pages ###

Shop base: #474 - /e-shop/
Cart: #475 - /cart/ -  Contains the [woocommerce_cart] shortcode
Checkout: #476 - /checkout/ -  Contains the [woocommerce_checkout] shortcode
My account: #477 - /my-account/
Terms and conditions: #301 - /terms-of-use/

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.19.3
Parent Theme Author URL: https://uxthemes.com
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart-shipping.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/cart/mini-cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome-child/woocommerce/checkout/form-pay.php
flatsome/woocommerce/checkout/order-received.php
flatsome-child/woocommerce/checkout/payment.php
flatsome/woocommerce/checkout/terms.php
flatsome-child/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product-cat.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/header.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/rating.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome-child/woocommerce/myaccount/form-edit-account.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome-child/woocommerce/myaccount/orders.php
flatsome-child/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome-child/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
reactify-classic-payments-settings

Daily Cron: ✔ Next scheduled: 2024-09-15 10:20:26 +03:00
Options: ✔
Notes: 67
Onboarding: completed

### Action Scheduler ###

Complete: 445
Oldest: 2024-09-09 15:43:02 +0000
Newest: 2024-09-15 00:13:31 +0000

Failed: 4
Oldest: 2024-09-11 14:16:04 +0000
Newest: 2024-09-13 09:52:54 +0000

Pending: 2
Oldest: 2024-09-15 15:45:06 +0000
Newest: 2024-09-16 00:13:31 +0000


### Status report information ###

Generated at: 2024-09-15 12:14:16 +03:00
```
</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by mikejolley** [comment]
This might be related to https://github.com/woocommerce/woocommerce/issues/34516 It's an issue with the shortcodes and products page so I'll assign to the correct team. Ideally the response via AJAX would return some notices which could then be served above the content.

**Comment by pmcpinto** [comment]
Thanks for the report. I know this has been open a while, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've taken a look and it seems that the bug is still valid. I'll mark this as ready for the engineering team to pick up.
